csharp/ql/lib/semmle/code/csharp/Preprocessor.qll

```diff
@@ -208,21 +208,40 @@ class HiddenLineDirective extends LineDirective {
   override string getAPrimaryQlClass() { result = "HiddenLineDirective" }
 }
 
+abstract private class NumericOrSpanLineDirective extends LineDirective {
```

Comment:
No `abstract` classes. So you will need to add a charpred
```
NumericOrSpanLineDirective() { directive_lines(this, [2, 3]) }
```

Reply:
No worries. I also now realized that the class is `private`, so `abstract` would probably have been OK. But let's keep it as is.